src/models.rs

```diff
@@ -568,3 +639,485 @@ pub struct Generic {
     pub spec: serde_json::Value,
     pub status: Option<serde_json::Value>,
 }
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(untagged)]
+pub enum NumberOrString {
```

Comment:
Initially seeing the name of this type I thought "why not use [Either](https://docs.rs/either/latest/either/enum.Either.html)?" (that being familiar from [Haskell](https://hackage.haskell.org/package/base-4.21.0.0/docs/Data-Either.html)).

Then looking at this implementation I'm thinking that maybe the naming should be a bit more specific like `ComputeUnit` perhaps? Also: Should this maybe implement `FromStr` ?

Reply:
I renamed it and also added FromStr, so now `"1024kB".parse::<ComputeUnit>()` works :)